criu/pie/restorer.c

```diff
@@ -318,6 +318,25 @@ static int restore_creds(struct thread_creds_args *args, int procfd, int lsm_typ
 	return 0;
 }
 
+/*
+ * This function is to find out the corresponding vma_entry
+ * of a given vma, the algorithm is to simply scan the whole
+ * vma entries inside the `args`. After getting vma_entry, we
+ * can judge whether this entry can be loaded on demand or not.
+ */
+static inline VmaEntry * lookup(void * pos, struct task_restore_args *args) 
+{
+	VmaEntry * vma_entry;
+	int i;
+	for (i = 0; i < args->vmas_n; i++) {
+		vma_entry = args->vmas + i;
+		if ((unsigned long)pos >= vma_entry->start && (unsigned long)pos < vma_entry->end) {
```

Comment:
why do you need to do type conversion here?

Reply:
`pos` is a pointer of type `void` and comparison between pointer and integer will throw an error.